orchestrator/test_runner/src/transaction_stress_test.rs

```diff
@@ -166,39 +176,47 @@ pub async fn transaction_stress_test(
 
     for denom in denoms {
         info!("Requesting batch for {}", denom);
-        let res = send_request_batch_tx(keys[0].validator_key, denom, (0f64,"".to_string()), &contact, 1.0)
-            .await
-            .unwrap();
+        let res = send_request_batch_tx(
+            keys[0].validator_key,
+            denom,
+            (0f64, "".to_string()),
+            &contact,
+            1.0,
+        )
+        .await
+        .unwrap();
         info!("batch request response is {:?}", res);
     }
 
-    let start = Instant::now();
-    let mut good = true;
-    while Instant::now() - start < TOTAL_TIMEOUT {
-        good = true;
-        for keys in user_keys.iter() {
-            let e_dest_addr = keys.eth_dest_address;
-            for token in erc20_addresses.iter() {
-                let bal = web30.get_erc20_balance(*token, e_dest_addr).await.unwrap();
-                if bal != send_amount.clone() {
-                    good = false;
+    match tokio::time::timeout(TOTAL_TIMEOUT, async {
+        loop {
+            good = true;
+            for keys in user_keys.iter() {
+                let e_dest_addr = keys.eth_dest_address;
+                for token in erc20_addresses.iter() {
+                    let bal = web30.get_erc20_balance(*token, e_dest_addr).await.unwrap();
+                    if bal != send_amount.clone() {
+                        good = false;
+                    }
                 }
             }
+            tokio::time::sleep(Duration::from_secs(5)).await;
```

Comment:
Same here, we still need a `break` condition if we want to exit `loop` cleanly.

Reply:
https://github.com/PeggyJV/gravity-bridge/pull/274/commits/a6cb54a1a973da56f9d3445187227c378ac3e86c